src/main/java/uk/gov/hmcts/ccd/data/caseaccess/CachedCaseRoleRepository.java

```diff
@@ -17,15 +15,15 @@
     public static final String QUALIFIER = "cached";
 
     private final CaseRoleRepository caseRoleRepository;
-    private final Map<String, Set<String>> caseRoles = newHashMap();
 
     public CachedCaseRoleRepository(@Qualifier(DefaultCaseRoleRepository.QUALIFIER)
                                     final CaseRoleRepository caseRoleRepository) {
         this.caseRoleRepository = caseRoleRepository;
     }
 
     @Override
+    @Cacheable("caseRolesForCaseTypeCache")
```

Comment:
I think normal Cacheable annotation may not work as expected with @RequestScope bean. You might need something like 
` @Cacheable(value = "caseRolesForCaseTypeCache", cacheManager = "requestScopedCacheManager")`

Reply:
As per our discussion, we have decided to go with the HashMap approach for @RequestScope cache beans.